core_impl/src/main/java/io/opencensus/implcore/stats/MutableAggregation.java

```diff
@@ -38,6 +39,18 @@ private MutableAggregation() {
   abstract void add(double value);
 
   /**
+   * Creates a new {@code MutableAggregation} by combining the internal values of this {@code 
+   * MutableAggregation} and the given {@code MutableAggregation}, with the given fraction.
+   * 
+   * @param other the other {@code MutableAggregation}. The type of this and other {@code
+   *     MutableAggregation} must match.
+   * @param fraction the fraction that the value in other {@code MutableAggregation} should
+   *     contribute. Must be within [0.0, 1.0].
+   * @return a new {@code MutableAggregation} with the combined value.
+   */
+  abstract MutableAggregation combine(MutableAggregation other, double fraction);
```

Comment:
It looks like this method is only used to create a ViewData when the stats are queried.  In that case, it seems like you could take advantage of the class' mutability to avoid returning a new MutableAggregation.  You could create an empty MutableAggregation for each Aggregation/TagValue combination and then call `combine` on the new MutableAggregation to add each time bucket to it.

Reply:
Nice suggestion, actually this method is only used to add fractional values to an empty MutableAggregation. I've changed this method to return void and add values to its internal value instead.